metals/src/main/scala/scala/meta/internal/metals/formatting/OnTypeFormattingProvider.scala

```diff
@@ -0,0 +1,74 @@
+package scala.meta.internal.metals.formatting
+
+import scala.meta.internal.metals.Buffers
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.metals.UserConfiguration
+import scala.meta.internal.parsing.Trees
+import scala.meta.tokens.Tokens
+
+import org.eclipse.lsp4j.DocumentOnTypeFormattingParams
+import org.eclipse.lsp4j.Position
+import org.eclipse.lsp4j.Range
+import org.eclipse.lsp4j.TextEdit
+import scala.meta.inputs.Input
+
+class OnTypeFormatter {
+  def contribute(
+      onTypeformatterParams: OnTypeFormatterParams
+  ): Option[List[TextEdit]] = None
+
+}
+
+case class OnTypeFormatterParams(
+    sourceText: String,
+    position: Position,
+    triggerChar: String,
+    startPos: meta.Position,
+    endPos: meta.Position,
+    tokens: Option[Tokens]
+) {
+  lazy val splitLines: Array[String] = sourceText.split("\\r?\\n")
+  val range = new Range(position, position)
+}
+
+class OnTypeFormattingProvider(
+    buffers: Buffers,
+    trees: Trees,
+    userConfig: () => UserConfiguration
+) {
+
+  // The order of which this is important to know which will first return the Edits
+  val formatters: List[OnTypeFormatter] = List(
+    MultilineString(userConfig)
+  )
+
+  def format(
+      params: DocumentOnTypeFormattingParams
+  ): List[TextEdit] = {
+    val uri = params.getTextDocument.getUri.toAbsolutePath
+    val range = new Range(params.getPosition, params.getPosition)
+    val triggerChar = params.getCh
+    val position = params.getPosition
+    buffers
+      .get(uri)
+      .map { sourceText =>
+        val virtualFile = Input.VirtualFile(sourceText.toString(), sourceText)
```

Comment:
Instead of using `contribute` method I just inlined everything and removed the parent class. This was just adding to the cognitive overload.